pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -377,6 +379,44 @@ func (configgen *ConfigGeneratorImpl) buildGatewayHTTPRouteConfig(node *model.Pr
 	return routeCfg
 }
 
+func collapseDuplicateRoutes(input map[host.Name]*route.VirtualHost) map[host.Name]*route.VirtualHost {
+	dedupe := make(map[host.Name]*route.VirtualHost, len(input))
+	for h, vh := range input {
```

Comment:
Need to change this rfom n^2 to 2n ideally. It can be done if we can turn .Routes list into a hashable key for a map I think